app/Http/Controllers/Frontend/FlightController.php

```
@@ -268,4 +275,84 @@ public function show(string $id): View
+    private function CountAvailableAircraftForFlights($flights): array
+    {
+        $counts = [];
+        $batchSize = 50;
+
+        foreach ($flights->chunk($batchSize) as $batch) {
+            // Collect unique airports and subfleets
+            $airportIds = $batch->pluck('dpt_airport_id')->unique()->toArray();
+            $subfleetIds = collect($batch)->flatMap(fn($f) => $f->subfleets->pluck('id'))->unique()->toArray();
+
+            // Get airline IDs for this batch
+            $airlineIds = collect($batch)->pluck('airline_id')->unique()->toArray();
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$airlineIds'. (undefined)

(UnusedLocalVariable)